### PR TITLE
some refactor in ip change notifier source code

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: src/ip_notifier.cpp [hunk 1]
```diff
@@ -53,17 +55,21 @@ namespace libtorrent
...
 #elif defined TORRENT_WINDOWS
 		if (!m_hnd.is_open())
+#ifndef BOOST_NO_EXCEPTIONS
 			throw system_error(WSAGetLastError(), system_category());
+#else
+			std::terminate();
+#endif // BOOST_NO_EXCEPTIONS
 		m_ovl.hEvent = m_hnd.native_handle();
-#elif !TORRENT_USE_NETLINK
+#else
```
Comment (ssiloti) [inline src/ip_notifier.cpp]: This should not be changed. I don't want the TORRENT_UNUSED line on Linux, it is only for unsupported platforms.

File: src/ip_notifier.cpp [hunk 2]
```diff
@@ -53,17 +55,21 @@ namespace libtorrent
...
 		TORRENT_UNUSED(ios);
 #endif
 	}
 
 	ip_change_notifier::~ip_change_notifier()
 	{
-#if defined TORRENT_WINDOWS && !defined TORRENT_BUILD_SIMULATOR
 		cancel();
+#if defined TORRENT_WINDOWS && !defined TORRENT_BUILD_SIMULATOR
```
Comment (ssiloti) [inline src/ip_notifier.cpp]: This should not be changed. The cancel call is only needed on Windows.

File: src/ip_notifier.cpp (hunk 1 above)
Comment (aldenml) [inline src/ip_notifier.cpp]: ok

File: src/ip_notifier.cpp (hunk 2 above)
Comment (aldenml) [inline src/ip_notifier.cpp]: not convinced but can't discuss about it, I will revert the line

File: include/libtorrent/ip_notifier.hpp [hunk 3]
```diff
@@ -7,14 +7,14 @@ Redistribution and use in source and binary forms, with or without
...
 
-* Redistributions of source code must retain the above copyright
-notice, this list of conditions and the following disclaimer.
-* Redistributions in binary form must reproduce the above copyright
-notice, this list of conditions and the following disclaimer in
-the documentation and/or other materials provided with the distribution.
-* Neither the name of the author nor the names of its
-contributors may be used to endorse or promote products derived
-from this software without specific prior written permission.
+    * Redistributions of source code must retain the above copyright
```
Comment (arvidn) [inline include/libtorrent/ip_notifier.hpp]: I take it this is the formatting used in all other files, right?

File: include/libtorrent/netlink.hpp [hunk 4]
```diff
@@ -98,7 +95,7 @@ namespace libtorrent
 
 		const data_type* data() const
 		{
-			return (struct sockaddr*)&sockaddr;
```
Comment (arvidn) [inline include/libtorrent/netlink.hpp]: I would have expected clang or gcc on travis to warn on this... hm

File: include/libtorrent/netlink.hpp [hunk 5]
```diff
@@ -158,9 +155,7 @@ namespace libtorrent
 		using endpoint = basic_nl_endpoint<netlink>;
 		using socket = boost::asio::basic_raw_socket<netlink>;
 
-		netlink() : nl_family(0)
-		{
-		}
+		netlink() : netlink(0) {}
```
Comment (arvidn) [inline include/libtorrent/netlink.hpp]: there isn't a name for family 0?

File: src/ip_notifier.cpp (hunk 2 above)
Comment (arvidn) [inline src/ip_notifier.cpp]: maybe there should be a comment describing why it's only necessary on windows

File: include/libtorrent/ip_notifier.hpp (hunk 3 above)
Comment (aldenml) [inline include/libtorrent/ip_notifier.hpp]: yes, just formatting

File: include/libtorrent/netlink.hpp (hunk 5 above)
Comment (aldenml) [inline include/libtorrent/netlink.hpp]: let me check

Comment (gubatron) [inline include/libtorrent/netlink.hpp]: `#define AF_UNSPEC   0` maybe?

**Pull request merged**